src/main/perl/FileWriter.pm

```diff
@@ -215,21 +231,43 @@ sub close
         *$self->{save} = 0;
         my $content_ref = *$self->{buf};
 
+        if (*$self->{original_from_source}) {
+            # Report changes compared to source
+            my $src_original_content = *$self->{original_content};
+            if (defined($src_original_content)) {
+                my $src_diff = diff(\$src_original_content, $content_ref, { STYLE => "Unified" });
+                my $src_changed = $src_diff ? 1 : 0;
+                if ($src_changed) {
+                    if(*$self->{options}->{sensitive}) {
+                        $self->verbose("Changes compared to source for $filename are not reported due to sensitive content");
+                    } else {
+                        $self->verbose("Changes compared to source for $filename:");
+                        $self->report($src_diff) if $self->is_verbose();
+                    }
+                } else {
+                    $self->verbose("No changes compared to source for $filename");
+                }
+            } else {
+                $self->verbose("No original source content for $filename");
+            }
+        }
+
         # Always read and (try to) determine the diff
-        my $content_orig;
-        if (defined(*$self->{content_orig})) {
-            $self->debug(2, "Using original contents from $filename");
-            $content_orig = *$self->{content_orig};
+        my $original_content;
```

Comment:
hmmm, i'd rather not. the previous block is only about reporting, this block is about setting the content to use to determine change. i don't want to mix these, this is the critical bit of the close method.
maybe the attribute names can be changed?
i can also add that the first block is the "normal" path for FileEditor w/o source; the second path is the normal for FileWriter and FileEditor w source.
i just added the role and meaning of the attributes to the pod to make it clear what happens, i don't assume people should use them directly. maybe i should move the last block in the pod to regular comment?